src/ray/raylet/scheduling_queue.cc

```diff
@@ -2,6 +2,45 @@
 
 #include "ray/status.h"
 
+namespace {
+
+// Helper function to remove tasks in the given set of task_ids from a
+// queue, and append them to the given vector removed_tasks.
+void RemoveTasksFromQueue(std::list<ray::raylet::Task> &queue,
+                          std::unordered_set<ray::TaskID> &task_ids,
+                          std::vector<ray::raylet::Task> &removed_tasks) {
+  for (auto it = queue.begin(); it != queue.end();) {
+    auto task_id = task_ids.find(it->GetTaskSpecification().TaskId());
+    if (task_id != task_ids.end()) {
+      task_ids.erase(task_id);
+      removed_tasks.push_back(std::move(*it));
+      it = queue.erase(it);
+    } else {
+      it++;
+    }
+  }
+}
```

Comment:
I've had to do awkward things with the scheduling queues before, not sure what the best way to fix this is, but some kinds of queries are a bit hard to do, e.g.,

https://github.com/ray-project/ray/blob/35f4a3070c15a516c8fd30b457ded1678b93a835/src/ray/raylet/node_manager.cc#L384-L391

Reply:
Yeah, until we have a way to do fast lookup of a task's state, I think this is the best option for now...

For that snippet that you posted, maybe it makes sense to just remove the task from the queue? I don't know if we should be leaving that task in the running tasks queue anyway.

Reply:
Oh good point, I commented about this above.